jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/Utf8Appendable.java

```diff
@@ -305,20 +299,20 @@ public void appendByte(byte b) throws IOException
                         }
                     }
                     _codep = 0;
-                    _state = current;
+                    _state = next;
```

Comment:
BTW, this code is from http://bjoern.hoehrmann.de/utf-8/decoder/dfa/ (see javadoc)

The name "current" is the name used in the original code.
As is the decode() method.